src/metabase/query_processor/expand.clj

```diff
@@ -307,7 +307,7 @@
       :next    (recur f  1 unit))
     (let [f (datetime-field f unit)]
       (cond
-        (core/= n  0) (= f (value f (relative-datetime :current)))
+        (core/= n  0) (= f (value f (relative-datetime  0 unit)))
```

Comment:
any are you sure that doesn't break anything for the other query processors, like SQL